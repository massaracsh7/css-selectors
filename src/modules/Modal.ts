import Button from './Button';
import Block from './Block';
import Element from './Element';

export default class Modal {
  create = (): HTMLElement => {
    const mess = new Element('div', 'modal__text', 'Game over');
    const btn = new Button('modal__btn', 'New Game', 'button', this.show);
    const modalWrapper = new Block('div', 'modal__wrapper', mess.create(), btn.create());
    const modal = new Block('div', 'modal', modalWrapper.create());
    return modal.create();
  };

  show = (): void => {
    const modal2 = document.querySelector('.modal');
    modal2?.classList.remove('modal--active');
    localStorage.clear();
    window.location.reload();
  };
}
