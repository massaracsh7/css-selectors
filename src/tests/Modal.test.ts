/** @type {import('<rootDir>/node_modules/ts-jest/dist/types').InitialOptionsTsJest} */

import Modal from '../modules/Modal';

describe('Modal functionality', () => {
  let modal2: HTMLElement;
  let originalLocation: Location;

  beforeEach(() => {
    // Set up a new modal instance and append it to the body
    modal2 = document.createElement('div');
    modal2.classList.add('modal');
    modal2.classList.add('modal--active');
    document.body.append(modal2);

    // Preserve the original location object
    originalLocation = window.location;
    delete (window as any).location;
    (window as any).location = { reload: jest.fn() };
  });

  afterEach(() => {
    // Clean up the modal from the body
    modal2.remove();

    // Restore the original location object
    (window as any).location = originalLocation;
  });

  describe('create modal', () => {
    it('should return modal and match snapshot', () => {
      const modal = new Modal().create();
      expect(modal).toMatchSnapshot();
    });
  });

  describe('load new game', () => {
    it('should reload window', () => {
      const modal = new Modal();
      modal.show();
      expect((window as any).location.reload).toHaveBeenCalled();
    });

    it('should remove class modal--active', () => {
      const modal = new Modal();
      modal.show();
      expect(modal2.classList.contains('modal--active')).toBe(false);
    });
  });
});
