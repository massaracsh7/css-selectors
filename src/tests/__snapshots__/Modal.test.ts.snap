// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal functionality create modal should return modal and match snapshot 1`] = `
<div
  class="modal"
>
  <div
    class="modal__wrapper"
  >
    <div
      class="modal__text"
    >
      Game over
    </div>
    <button
      class="modal__btn"
      type="button"
    >
      New Game
    </button>
  </div>
</div>
`;
